front-end/src/app/shared/components/transaction-type-base/transaction-type-base.component.spec.ts

```diff
@@ -304,7 +304,20 @@ describe('TransactionTypeBaseComponent', () => {
       life: 3000,
     };
     const messageServiceAddSpy = spyOn(testMessageService, 'add');
-    component.navigateTo(new NavigationEvent(NavigationAction.SAVE, NavigationDestination.ANOTHER));
+    spyOn(testRouter, 'navigateByUrl').and.callFake(
+      () =>
+        new Promise((res) => {
```

Comment:
SonarCloud is flagging the new Promise() as a code smell. Take a look and see if it can be easily changed to get the smell out.

Reply:
This was my code, that I shared with Elaine 😳